File: mycollab-web/src/main/java/com/esofthead/mycollab/module/file/events/FileEvent.java
package com.esofthead.mycollab.module.file.events;

import com.esofthead.mycollab.eventmanager.ApplicationEvent;

public class FileEvent {
	public static class GotoList extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoList(Object source, Object data) {
			super(source, data);
		}
	}
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/file/events/FileEvent.scala
package com.esofthead.mycollab.module.file.events

import com.esofthead.mycollab.eventmanager.ApplicationEvent
import com.esofthead.mycollab.module.ecm.domain.{ExternalDrive, Resource}

/**
  * @author MyCollab Ltd
  * @since 5.1.1
  */
object FileEvent {

  class GotoList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class ResourceRemovedEvent(source: AnyRef, data: Resource) extends ApplicationEvent(source, data) {}

  class ExternalDriveConnectedEvent(source: AnyRef, data: ExternalDrive) extends ApplicationEvent(source, data) {}

  class ExternalDriveDeleteEvent(source: AnyRef, data: ExternalDrive) extends ApplicationEvent(source, data) {}

}
